extensions/histogram/src/main/java/io/druid/query/aggregation/histogram/ApproximateHistogramAggregatorFactory.java

```diff
@@ -115,6 +116,36 @@ public AggregatorFactory getCombiningFactory()
   }
 
   @Override
+  public AggregatorFactory getMergingFactory(AggregatorFactory other) throws AggregatorFactoryNotMergeableException
+  {
+    if (other.getName().equals(this.getName()) && other instanceof ApproximateHistogramAggregatorFactory) {
+      ApproximateHistogramAggregatorFactory castedOther = (ApproximateHistogramAggregatorFactory) other;
+
+      if (Float.compare(castedOther.lowerLimit, lowerLimit) == 0 &&
+          numBuckets != castedOther.numBuckets &&
+          resolution != castedOther.resolution &&
```

Comment:
Am I reading this right? If the `numBuckets`, `resolution`, or `upperLimit` are different then you can't merge two approximate histograms?

Also, as I understand them,

- the numBuckets is only related to finalization, and shouldn't be relevant for merging.
- you actually can merge two histograms with different resolutions and limits.
- probably you want the merged upperlimit to be the max of the two, the merged lowerlimit to be the min of the two, and the merged resolution to be the max of the two.